src/create_maps.cpp

```diff
@@ -0,0 +1,366 @@
+
+//Data for the Second chunk	
+    for (int i=chunk_size;i<posSize;i++){
+	for(int j=0;j<PY;j++){
+	   
+	   dset1_data[i-chunk_size][j]=poseReach[i][j];
+	   }
+     } 
+    
+   
+//Resizing new memory dataspace indicating new size of the buffer
+    dims[0] = posSize-chunk_size;
+    dims[1] = ncols;
+    H5Sset_extent_simple(mem_space, ndims, dims, NULL);
+    
+
+//Extend dataset
+    dims[0] = posSize;
+    dims[1] = ncols;
+    H5Dset_extent(dset, dims);
+    
+
+//Selecting hyperslab
+    file_space = H5Dget_space(dset);
+    start[0] = chunk_size;
+    start[1] = 0;
+    count[0] = posSize-chunk_size;
+    count[1] = ncols;
+    H5Sselect_hyperslab(file_space, H5S_SELECT_SET, start, NULL, count, NULL);
+    
+
+//Writing buffer to dataset
+    H5Dwrite(dset, H5T_NATIVE_FLOAT, mem_space, file_space, H5P_DEFAULT, buffer);
+   
+//Closing all the resources
+    delete[] dset1_data;
+    delete[] buffer;
+    H5Sclose(file_space);
+    H5Sclose(mem_space);
+    H5Dclose(dset);
+    H5Gclose(group_poses);
+        
+    
+
+//Creating Sphere dataset
+    ROS_INFO("Saving spheres in Reachability map");
+    hid_t reachability_sphere_dataset, reachability_sphere_dataspace;
+    const int SX = sphereColor.size();                     
+    const int SY = 4;
+    
+    hsize_t dims2[2];               // dataset dimensions
+    dims2[0] = SX;
+    dims2[1] = SY;
+    double dset2_data[SX][SY];
+
+    for (map<vector<double>, double> ::iterator it = sphereColor.begin();it != sphereColor.end();++it){
+        for(int j=0;j<SY-1;j++){
+	    dset2_data[distance(sphereColor.begin(), it)][j]=it->first[j];
+	    }
+        for(int j=3;j<SY;j++){
+	    dset2_data[distance(sphereColor.begin(), it)][j]=it->second;
+	    }
+    }
+    reachability_sphere_dataspace = H5Screate_simple(2, dims2, NULL);
+    reachability_sphere_dataset = H5Dcreate2(group_spheres,"sphere_dataset",H5T_NATIVE_DOUBLE, reachability_sphere_dataspace,  H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+    
+    H5Dwrite(reachability_sphere_dataset, H5T_NATIVE_DOUBLE, H5S_ALL, H5S_ALL, H5P_DEFAULT, 
+                     dset2_data);
+
+    
+    
+//Creating attribute
+    
+    hid_t attr_id;
+    hsize_t attr_dims;
+    float attr_data[1];
+    attr_data[0] = resolution;
+    attr_dims =1;
+    reachability_sphere_dataspace = H5Screate_simple(1, &attr_dims, NULL);
+    attr_id = H5Acreate2 (reachability_sphere_dataset, "Resolution", H5T_NATIVE_FLOAT, reachability_sphere_dataspace, H5P_DEFAULT, H5P_DEFAULT);
+    H5Awrite(attr_id, H5T_NATIVE_FLOAT, attr_data);
+    H5Aclose(attr_id);
+    
+
+//Closing all
+
+    H5Sclose(reachability_sphere_dataspace);
+    H5Dclose(reachability_sphere_dataset);
+    H5Gclose(group_spheres);
+    H5Gclose(group_reachability);
+    H5Fclose(file);
+    time (&finish);
+    double dif = difftime (finish,startit);
+    ROS_INFO ("Elasped time is %.2lf seconds.", dif );
+    ROS_INFO ("Completed");
+    ros::spinOnce();
+    //sleep(10000);
+    return -1;
```

Comment:
:) its a big mistake. Sorry about that. Updated that. Actually it was making maps in loop.